### ISSUE TITLE
nix copy —to s3:// fails with aws-sdk-cpp 1.9.121

### ISSUE THREAD

**Issue by neosimsim** [NixOS/nix#5404]
When I update nix I can't copy to our binary cache anymore. All I get is

```
error: don't know how to open Nix store 's3://my-cache'
```

I bisected nixpkgs and found that the bug was introduced by https://github.com/NixOS/nixpkgs/commit/abbb7c6e60b60f2a516b8a69c45c5b8bb415cc0f. @r-burns further broke it down to https://github.com/aws/aws-sdk-cpp/commit/cd509e997bb49811517381d941e2ab57d81f9067

### Steps To Reproduce
Run

```
$ nix shell github:NixOS/nixpkgs?rev=abbb7c6e60b60f2a516b8a69c45c5b8bb415cc0f#nix --command nix copy --to 's3://my-cache' 
error: don't know how to open Nix store 's3://my-cache'
```

The parent commit works:

```
$ nix shell github:NixOS/nixpkgs?rev=4786f9e8453cf5a632a91c14a1940ad5029735a7#nix --command nix copy --to 's3://my-cache' 
error: AWS error fetching 'nix-cache-info': Unable to parse ExceptionName: PermanentRedirect Message: The bucket you are attempting to access must be addressed using the specified endpoint. Please send all future requests to this endpoint.
```

This error here is fine because our cache has a different name of course.

I also reported an error at https://github.com/NixOS/nixpkgs/issues/142107.

**Comment by neosimsim** [comment]
Fixed by [#142300](https://github.com/NixOS/nixpkgs/pull/142306)

**Comment by mibmo** [comment]
Seems this is happening again on `2.24.0pre20240723_fb450de2` (`nixVersions.git` on nixpkgs unstable as of 2024-07-29)?
